pkg/controller/build/imagebuilder/podimagebuilder.go

```diff
@@ -0,0 +1,210 @@
+		return nil, err
+	}
+
+	mosbName, err := p.getMachineOSBuildName()
+	if err != nil {
+		return nil, err
+	}
+
+	buildPod := builder.GetObject().(*corev1.Pod)
+
+	bp, err := p.kubeclient.CoreV1().Pods(ctrlcommon.MCONamespace).Create(ctx, buildPod, metav1.CreateOptions{})
+	if err == nil {
+		klog.Infof("Build pod %q created for MachineOSBuild %q", bp.Name, mosbName)
+		return bp, nil
+	}
+
+	if k8serrors.IsAlreadyExists(err) {
+		return p.getBuildPodStrict(ctx)
+	}
+
+	return nil, fmt.Errorf("could not create build pod: %w", err)
+}
+
+// Gets the build pod, returning any errors in the process.
+func (p *podImageBuilder) getBuildPodStrict(ctx context.Context) (*corev1.Pod, error) {
+	// TODO: Use a lister for this instead.
+	return p.kubeclient.CoreV1().Pods(ctrlcommon.MCONamespace).Get(ctx, p.getBuilderName(), metav1.GetOptions{})
+}
+
+// Gets the build pod but returns nil if it is not found.
+func (p *podImageBuilder) getBuildPod(ctx context.Context) (*corev1.Pod, error) {
+	pod, err := p.getBuildPodStrict(ctx)
+	if err == nil {
+		return pod, nil
+	}
+
+	if k8serrors.IsNotFound(err) {
+		return nil, nil
+	}
+
+	return nil, err
+}
+
+// Gets the MachineOSBuildStatus for the currently running build.
+func (p *podImageBuilder) MachineOSBuildStatus(ctx context.Context) (mcfgv1alpha1.MachineOSBuildStatus, error) {
+	status, err := p.machineOSBuildStatus(ctx)
+	if err != nil {
+		return status, p.addMachineOSBuildNameToError(fmt.Errorf("could not get MachineOSBuildStatus: %w", err))
+	}
+
+	return status, nil
+}
+
+func (p *podImageBuilder) getStatus(ctx context.Context) (*corev1.Pod, mcfgv1alpha1.BuildProgress, []metav1.Condition, error) {
+	pod, err := p.getBuildPodStrict(ctx)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	status, conditions, err := p.mapPodStatusToBuildStatus(pod)
+	if err != nil {
+		return nil, "", nil, fmt.Errorf("could not map build pod status to buildstatus: %w", err)
+	}
+
+	klog.Infof("Build pod %q phase %q mapped to MachineOSBuild progress %q", pod.Name, pod.Status.Phase, status)
+
+	return pod, status, conditions, nil
+}
+
+func (p *podImageBuilder) machineOSBuildStatus(ctx context.Context) (mcfgv1alpha1.MachineOSBuildStatus, error) {
+	pod, status, conditions, err := p.getStatus(ctx)
+	if err != nil {
+		return mcfgv1alpha1.MachineOSBuildStatus{}, err
+	}
+
+	return p.getMachineOSBuildStatus(ctx, pod, status, conditions)
+}
+
+// Gets only the build progress field for a currently running build.
+func (p *podImageBuilder) Status(ctx context.Context) (mcfgv1alpha1.BuildProgress, error) {
+	_, status, _, err := p.getStatus(ctx)
+	if err != nil {
+		return status, p.addMachineOSBuildNameToError(fmt.Errorf("could not get BuildProgress: %w", err))
+	}
+
+	return status, nil
+}
+
+// Interrogates the pod object and maps each phase to a MachineOSBuild status.
+func (p *podImageBuilder) mapPodStatusToBuildStatus(pod *corev1.Pod) (mcfgv1alpha1.BuildProgress, []metav1.Condition, error) {
+	switch pod.Status.Phase {
```

Comment:
non-blocking question:
How does the build get into `interrupted` status

Reply:
I think the interrupted status is a holdover from before I had the insight about using Jobs instead of bare pods to do the build. I feel like it might not be needed anymore, at least not in this context.

Reply:
Just to follow up on this point: `Interrupted` means that the underlying build pod for a given MachineOSBuild was deleted. This should be considered a terminal state for the MachineOSBuild where it cannot transition from that state into, say, running, or successful.